### PR TITLE
#897 - Cursor pagination does not preserve order with the "search" where arg

### PR DESCRIPTION
**Pull request by jasonbahl**
What does this implement/fix? Explain your changes.
---------------------------------------------------
WP_Query applies default ordering by title when the "s" arg is passed to it, which was causing funky issues with WPGraphQL Cursor Pagination.

This sets the `search_orderby_title` to false, and defaults to orderby date. Custom orderby options can be passed to the connection as well.

Does this close any currently open issues?
------------------------------------------
#897 

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
To test, I created 4 posts with the word "floof" as the content. 

In the admin, you can see a search for "floof" shows posts: `Four, Three, Two, One"

Screen Shot 2019-07-09 at 2 13 50 PM

### Query the last 2

Here, I query for the `last: 2` items, and as expected, we get Posts `Two` and `One`.

#### Query
```graphql
query GET_POSTS($first: Int, $last: Int, $after: String, $before: String, $where: RootQueryToPostConnectionWhereArgs) {
  posts(last: $last, before: $before, first: $first, after: $after, where: $where) {
    pageInfo {
      hasPreviousPage
      hasNextPage
      startCursor
      endCursor
    }
    edges {
      cursor
      node {
        id
        date
        title
      }
    }
  }
}
```

#### Variables
```json
{
  "last": 2,
  "before": "YXJyYXljb25uZWN0aW9uOjEzOTc=",
  "where": {
    "search": "floof"
  }
}
```

#### Screenshot of results
Screen Shot 2019-07-09 at 2 15 29 PM

### Query with before cursor

Now, if we execute the same query, but use the `startCursor` from the previous query results as the value for the `before` input we get items `Four` and `Three`:

#### Variables
```json
{
  "last": 2,
  "before": "YXJyYXljb25uZWN0aW9uOjEzOTc=",
  "where": {
    "search": "floof"
  }
}
```

#### Screenshot of paginated query
Screen Shot 2019-07-09 at 2 17 54 PM

Any other comments
---------------------------
Tests are included for forward and backward pagination using the `{where: { search: ""} }` argument

Where has this been tested?
---------------------------
**Operating System:** Mac OSX 10.14.5
**WordPress Version:** WordPress v5.2.2

### CODE REVIEW

File: src/Data/Cursor/PostObjectCursor.php [hunk 1]
```diff
@@ -121,6 +121,7 @@ public function get_where() {
 		$orderby = $this->get_query_var( 'orderby' );
 		$order   = $this->get_query_var( 'order' );
 
+
```
Comment (hughdevore) [inline src/Data/Cursor/PostObjectCursor.php]: 👀

File: src/Data/Cursor/PostObjectCursor.php [hunk 2]
```diff
@@ -129,6 +130,7 @@ public function get_where() {
 				$this->compare_with( $by, $order );
 			}
 		} else if ( ! empty( $orderby ) && is_string( $orderby ) ) {
+
```
Comment (hughdevore) [inline src/Data/Cursor/PostObjectCursor.php]: 👀

File: src/Data/Cursor/PostObjectCursor.php (hunk 1 above)
Comment (hughdevore) [inline src/Data/Cursor/PostObjectCursor.php]: 👀

File: tests/wpunit/PostObjectSearchTest.php [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+		]);
+
+		$this->created_post_ids = $this->create_posts();
+
+		$this->app_context = new \WPGraphQL\AppContext();
+
+		$this->app_info = new \GraphQL\Type\Definition\ResolveInfo( array() );
+
+		$this->query = '
+		query GET_POSTS($first: Int, $last: Int, $after: String, $before: String $where:RootQueryToPostConnectionWhereArgs) {
```
Comment (let00) [inline tests/wpunit/PostObjectSearchTest.php]: @jasonbahl  seems like a comma is missing before the where arg

Comment (jasonbahl) [reply tests/wpunit/PostObjectSearchTest.php]: good call. Commas in variable declarations are optional, so not a big deal, but probably should adjust for consistency 😄

**Pull request merged**